modules/vmfcore/java/Variant.java

```diff
@@ -0,0 +1,219 @@
+ package com.intel.vmf;
+ 
+ public class Variant
+ {
+    static
+    {
+        System.loadLibrary("vmf");
+    }
+	
+    public enum Type
+    {
+    	type_unknown,
+        type_char,
+        type_integer,
+        type_real,
+        type_string,
+        type_vec2d,
+        type_vec3d,
+        type_vec4d,
+        type_rawbuffer,
+        type_char_vector,
+        type_integer_vector,
+        type_real_vector,
+        type_string_vector,
+        type_vec2d_vector,
+        type_vec3d_vector,
+        type_vec4d_vector
+    }
+    
+    protected final long nativeObj;
+	
+    private Variant (long addr)
+    {
+        if (addr == 0)
+            throw new java.lang.UnsupportedOperationException("Native object address is NULL");
+        nativeObj = addr;
+    }
+    
+	public Variant ()
+	{
+	    nativeObj = n_Variant ();
+	}
+	
+	public Variant ( final int value )
+	{
+		nativeObj = n_Variant ( value );
```

Comment:
do we really need overloads for `n_Variant`?
isn't it better to name them differently?